de-lib/src/main/java/org/iplantc/de/diskResource/client/views/toolbar/DiskResourceViewToolbarImpl.java

```diff
@@ -283,46 +282,47 @@ public void onDiskResourceSelectionChanged(DiskResourceSelectionChangedEvent eve
         final boolean isSelectionInTrash = isSelectionInTrash(selectedDiskResources);
         final boolean isFolderSelect = !isSelectionEmpty
                                        && firstItem instanceof Folder;
+        final boolean containsFile = diskResourceUtil.containsFile(selectedDiskResources);
+        final boolean containsFolder = diskResourceUtil.containsFolder(selectedDiskResources);
 
         moveToTrashMiEnabled = !isSelectionEmpty && isOwner && !isSelectionInTrash && !containsFilteredItems;
 
         renameMiEnabled = !isSelectionEmpty && isSingleSelection && isOwner && !isSelectionInTrash && !containsFilteredItems;
         moveMiEnabled = !isSelectionEmpty && isOwner && !isSelectionInTrash && !containsFilteredItems;
         deleteMiEnabled = !isSelectionEmpty && isOwner && !containsFilteredItems;
         editFileMiEnabled = !isSelectionEmpty && isSingleSelection
-                && containsFile(selectedDiskResources) && isOwner && !isSelectionInTrash && !containsFilteredItems;
+                            && containsFile && isOwner && !isSelectionInTrash && !containsFilteredItems;
         editCommentsMiEnabled = !isSelectionEmpty && isSingleSelection && !isSelectionInTrash
                 && isReadable && !containsFilteredItems;
         editInfoTypeMiEnabled = !isSelectionEmpty && isSingleSelection && !isSelectionInTrash
-                && containsFile(selectedDiskResources) && isOwner && !containsFilteredItems;
+                                && containsFile && isOwner && !containsFilteredItems;
         metadataMiEnabled = !isSelectionEmpty && isSingleSelection && !isSelectionInTrash && !containsFilteredItems;
 
-        simpleDownloadMiEnabled = !isSelectionEmpty && containsFile(selectedDiskResources) && !containsFilteredItems;
+        simpleDownloadMiEnabled = !isSelectionEmpty && containsFile && !containsFilteredItems;
         bulkDownloadMiEnabled = !isSelectionEmpty && !containsFilteredItems;
         sendToCogeMiEnabled = !isSelectionEmpty
                 && isSingleSelection
-                && containsFile(selectedDiskResources)
+                && containsFile
                 && !isSelectionInTrash
                 && diskResourceUtil.isGenomeVizInfoType(getInfoTypeFromSingletonCollection(selectedDiskResources))
                 && !containsFilteredItems;
         sendToEnsemblMiEnabled = !isSelectionEmpty
-                && isSingleSelection
-                && containsFile(selectedDiskResources)
+                && !containsFolder
                 && !isSelectionInTrash
-                && diskResourceUtil.isEnsemblInfoType(getInfoTypeFromSingletonCollection(selectedDiskResources))
-                && !containsFilteredItems;
+                && !containsFilteredItems
+                && containsOnlyEnsemblFiles(selectedDiskResources);
```

Comment:
Here is the logic that enables the "View in Genome Browser" menu item when multiple genome files (or their index files) are selected, with the help of a new `containsOnlyEnsemblFiles` method.

Most of the other changes in this file were replacing the custom `containsFile` and `containsOnlyFolders` methods with booleans, so now we only have to call `diskResourceUtil.containsFile` and `diskResourceUtil.containsFolder` once, and use those results everywhere else in this method.